src/visbook/visbook.api.ts

```diff
@@ -0,0 +1,60 @@
+import { AccommodationAvailability } from './visbook.interface';
+
+export class VisbookApi {
+  async isCabinAvailable(
+    cabinVisbookId: number,
+    checkIn: string,
+    checkOut: string,
+  ): Promise<boolean> {
+    const visbookResponse = await this.makeRequest(
+      cabinVisbookId,
+      checkIn,
+      checkOut,
+    );
+
+    const accommodations = visbookResponse.accommodations;
+
+    if (accommodations === undefined) {
+      return false;
+    }
+
+    for (const accommodation of accommodations) {
+      if (accommodation.availability.available === true) return true;
+    }
+
+    return false;
```

Comment:
I think it is better from a separations of concerns perspective to move this logic to the visbook service.

`isCabinAvailable` is a business logic concept while this module is about interfacing with the Visbook API.

Reply:
Good point. I changed it to "getAccommodationAvailability" that prepares the request (to keep makeRequest private like you have it for the cabin api) and moved isCabinAvailable to the visbook service.